exploration/0475-overriding-extending-namespacing.md

```diff
@@ -0,0 +1,334 @@
+# Design Proposal Template
+
+Status: **Proposed**
+
+<details>
+	<summary>Metadata</summary>
+	<dl>
+		<dt>Contributors</dt>
+		<dd>@aphillips</dd>
+		<dt>First proposed</dt>
+		<dd>2023-09-13</dd>
+		<dt>Pull Request</dt>
+		<dd>#475</dd>
+	</dl>
+</details>
+
+## Objective
+
+_What is this proposal trying to achieve?_
+
+Implementations will provide the functionality for selection and formatting,
+including options and option values.
+Much of this functionality will be mandated by the default registry.
+We expect that default registry entries will serve the core needs for MF2 users.
+However, there are many capabilities available in platform, library,
+or operating environment APIs that could be useful to developers and translators
+or which might be expected on a specific platform or in a specific programming language.
+In addition, we expect to provide support for markup and templating regimes.
+These need to be implemented using values not found in the default registry.
+
+An additional hope is that a robust ecosystem of function libraries will be created.
+A successful ecosystem will allow users to pick-and-choose or cherry-pick fuctions or
+options to use in a given development environment.
+Each function, option, or option value extension needs to work as seamlessly as possible
+with other add-ons and with the built-in functionality.
+
+To that end, we need to define how externally-authored functions appear in a _message_;
```

Comment:
I think this paragraph should come immediately after the heading. As it is, the first sentence after `_What is this proposal trying to achieve?_` doesn't answer the question.

Reply:
Good point. Moved to start and slightly reworded for context.